contracts/src/LightClient.sol

```diff
@@ -373,6 +392,53 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
         }
     }
 
+    /// @notice updates the stateHistoryCommitments array each time a new
+    /// finalized state is added to the LightClient contract.
+    /// Ensures that the array contains only the most recent data and does not
+    /// exceed the maxStateHistoryDuration duration in seconds.
+    /// @dev the block timestamp is used to determine if the stateHistoryCommitments array
+    /// should be pruned, based on the maxStateHistoryDuration duration in seconds.
+    /// @dev a FIFO (First-In-First-Out) approach is used to delete elements from the start of the
+    /// array,
+    /// ensuring that only the most recent states are retained within the maxStateHistoryDuration
+    /// duration.
+    /// @dev the `delete` method does not reduce the array length but resets the value at the
+    /// specified index to zero.
+    /// the stateHistoryFirstIndex variable is used as an offset to indicate the starting point for
+    /// reading the array,
+    /// since the length of the array is not reduced even after deletion.
+    function updateStateHistory(
+        uint256 blockNumber,
+        uint256 blockTimestamp,
+        LightClientState memory state
+    ) internal {
+        if (maxStateHistoryDuration < 86400) revert InvalidMaxStateHistory();
```

Comment:
I think it's a bit unexpected to revert here because we already have a check in `setMaxStateHistoryDuration`.

Reply:
yes agreed

Reply:
https://github.com/EspressoSystems/espresso-sequencer/commit/e7800f0ac35db2b9458dffaaafacd09c6e12a6b5